cmd/entrypoint/runner_test.go

```diff
@@ -39,10 +41,48 @@ func TestRealRunnerStdoutAndStderrPaths(t *testing.T) {
 		stdoutPath: filepath.Join(tmp, "stdout"),
 		stderrPath: filepath.Join(tmp, "subpath/stderr"),
 	}
+
+	// capture the std{out/err} output to verify whether we print log in the std
+	oldStdout := os.Stdout // keep backup of the real stdout
+	outReader, outWriter, _ := os.Pipe()
+	os.Stdout = outWriter
+
+	oldStderr := os.Stderr
+	errReader, errWriter, _ := os.Pipe()
+	os.Stderr = errWriter
```

Comment:
This seems brittle since it modifies global state.

How about we:

1. Modify realRunner to take in io.Writers 
2. Have main.go initialize the multiwriters if the flag is set.
3. If the writers are nil, default to os.Stdout/Stderr

This way in tests we can just use a buffer and remove a bunch of the path manipulation?

Reply:
The test here is to verify that logs can be printed to standard output. If you modify realRunner to accept io.Writers, you need to pass in os.stdout, which is the same as the current situation.